lib/driver.js

```diff
@@ -412,10 +412,20 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
     } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
-      log.info(`Will reuse previously cached WDA instance at '${this.wda.url.href}'. ` +
-               `Set the wdaLocalPort capability to a value different from ${this.wda.url.port} ` +
-               `if this is an undesired behavior.`);
-      this.wda.webDriverAgentUrl = this.wda.url.href;
+      let currentAppBundleID;
+      if (util.hasValue(this.opts.updatedWDABundleId)) {
+        let response = await this.proxyCommand('/status', 'GET');
+        currentAppBundleID = response.currentApp.bundleID;
+      }
+      if (this.opts.updatedWDABundleId !== currentAppBundleID) {
+        log.info(`Will stop running WDA since the WDA has different bundle id. The WDA is '${currentAppBundleID}'.`);
+        this.wda.quit();
```

Comment:
await is missing

Reply:
Does `this.wda.quit()` do anything when we haven't started WDA? All we have is a running WDA process on the device, but `quit` essentially stops the xcodebuild process we have started. In this case there is no such xcodebuild process.

Reply:
I see... I stepped into the method and understand the method affects iproxy, xcodebuild  and jwproxy.